### PR TITLE
replace deprecated bind2nd with lambda expressions

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
@drwells, if the answer to checklist questions is "no" should I still check them off? :-)

**Comment by boyceg** [comment]
(E.g. I think that the answers to the middle questions are all "no".)

**Comment by boyceg** [comment]
I decided to go ahead an also get rid of `std::transform` in places where `std::for_each` seemed more appropriate.

File: src/IB/IBStandardForceGen.cpp [hunk 1]
```diff
@@ -228,56 +228,44 @@ IBStandardForceGen::initializeLevelData(const Pointer<PatchHierarchy<NDIM> > hie
...
+    std::for_each(d_beam_data[level_number].petsc_global_prev_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_global_prev_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+
+    std::for_each(d_target_point_data[level_number].petsc_node_idxs.begin(),
+                  d_target_point_data[level_number].petsc_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_target_point_data[level_number].petsc_global_node_idxs.begin(),
+                  d_target_point_data[level_number].petsc_global_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
```
Comment (drwells) [inline src/IB/IBStandardForceGen.cpp]: Could we rewrite this with an outer loop over vectors and then a range-for over their entries?

**Comment by drwells** [comment]
> @drwells, if the answer to checklist questions is "no" should I still check them off? :-)

Yup! If there is nothing to do for an item then we can just check if off immediately.

File: src/IB/IBStandardForceGen.cpp [hunk 2]
```diff
@@ -228,56 +228,22 @@ IBStandardForceGen::initializeLevelData(const Pointer<PatchHierarchy<NDIM> > hie
...
+                                                &d_beam_data[level_number].petsc_prev_node_idxs,
+                                                &d_beam_data[level_number].petsc_global_mastr_node_idxs,
+                                                &d_beam_data[level_number].petsc_global_next_node_idxs,
+                                                &d_beam_data[level_number].petsc_global_prev_node_idxs,
+                                                &d_target_point_data[level_number].petsc_node_idxs,
+                                                &d_target_point_data[level_number].petsc_global_node_idxs };
+    for (auto v_ptr : idx_vecs)
+    {
+        std::for_each(v_ptr->begin(), v_ptr->end(), [](int& i) { i *= NDIM; });
+    }
```
Comment (drwells) [inline src/IB/IBStandardForceGen.cpp]: isn't that so much nicer?

**Comment by boyceg** [comment]
@drwells can you easily run the test suite on a Linux system? :-)

**Comment by drwells** [comment]
The CI needs to learn how to run the test suite. That's on my todo list.

**Pull request merged**